workspace/resource_repo_test.go

```diff
@@ -0,0 +1,405 @@
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+
+	"github.com/databrickslabs/terraform-provider-databricks/qa"
+)
+
+func TestGetProviderFromUrl(t *testing.T) {
+	assert.Equal(t, "bitbucketCloud", getProviderFromUrl("https://user@bitbucket.org/user/repo.git"))
+	assert.Equal(t, "gitHub", getProviderFromUrl("https://github.com//user/repo.git"))
+	assert.Equal(t, "azureDevOpsServices", getProviderFromUrl("https://user@dev.azure.com/user/project/_git/repo"))
+	//	assert.Equal(t, "bitbucketCloud", getProviderFromUrl("https://user@bitbucket.org/user/repo.git"))
+	assert.Equal(t, "", getProviderFromUrl("https://abc/user/repo.git"))
+	assert.Equal(t, "", getProviderFromUrl("ewfgwergfwe"))
+}
+
+func TestResourceRepoRead(t *testing.T) {
+	repoID := 48155820875912
+	repoIdStr := fmt.Sprintf("%d", repoID)
+	url := "https://user@github.com/user/repo.git"
+	provider := "gitHub"
+	branch := "main"
+	path := "/Repos/Production/testrepo"
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   http.MethodGet,
+				Resource: fmt.Sprintf("/api/2.0/repos/%d", repoID),
+				Response: ReposResponse{
+					Id:           int64(repoID),
+					Url:          url,
+					Provider:     provider,
+					Branch:       branch,
+					HeadCommitId: "7e0847ede61f07adede22e2bcce6050216489171",
+					Path:         path,
+				},
+			},
+		},
+		Resource: ResourceRepo(),
+		Read:     true,
+		New:      true,
+		ID:       repoIdStr,
+	}.Apply(t)
+	assert.NoError(t, err, err)
+	assert.Equal(t, repoIdStr, d.Id())
+	assert.Equal(t, path, d.Get("path"))
+	assert.Equal(t, branch, d.Get("branch"))
+	assert.Equal(t, provider, d.Get("git_provider"))
+	assert.Equal(t, url, d.Get("url"))
+	assert.Equal(t, "7e0847ede61f07adede22e2bcce6050216489171", d.Get("commit_hash"))
+}
+
+func TestResourceRepoRead_NotFound(t *testing.T) {
+	repoID := 48155820875912
+	qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   http.MethodGet,
+				Resource: fmt.Sprintf("/api/2.0/repos/%d", repoID),
+				Response: common.APIErrorBody{
+					ErrorCode: "RESOURCE_DOES_NOT_EXIST",
+					Message:   "Repo could not be found",
+				},
+				Status: 404,
+			},
+		},
+		Resource: ResourceRepo(),
+		Read:     true,
+		Removed:  true,
+		ID:       fmt.Sprintf("%d", repoID),
+	}.Apply(t)
+}
+
+func TestResourceRepoDelete(t *testing.T) {
+	repoID := 48155820875912
+	repoIdStr := fmt.Sprintf("%d", repoID)
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   http.MethodDelete,
+				Resource: fmt.Sprintf("/api/2.0/repos/%d", repoID),
+				Status:   http.StatusOK,
+			},
+		},
+		Resource: ResourceRepo(),
+		Delete:   true,
+		ID:       repoIdStr,
+	}.Apply(t)
+	assert.NoError(t, err, err)
+	assert.Equal(t, repoIdStr, d.Id())
+}
+
+func TestResourceRepoCreateNoBranch(t *testing.T) {
```

Comment:
Can you run "make fmt" or how did you manage to get that two-space identation? :)

Reply:
I did run it, interesting how it got formatted that way

Reply:
ah, it was github mobile app